packages/gestalt/src/SlimBanner.js

```diff
@@ -176,68 +176,97 @@ export default function SlimBanner({
-              color={iconColor}
-              icon={icon}
-              size={16}
-            />
-          </Flex.Item>
-        )}
+  const startIcon =
+    !isDefault && iconAccessibilityLabel ? (
+      <Flex.Item alignSelf={shouldShowButtons ? undefined : 'start'}>
+        <Icon accessibilityLabel={iconAccessibilityLabel} color={iconColor} icon={icon} size={16} />
+      </Flex.Item>
+    ) : null;
 
-        <Flex.Item flex="grow">
-          <Box dangerouslySetInlineStyle={{ __style: !isDefault ? { marginTop: '-1px' } : {} }}>
-            {typeof message === 'string' ? (
-              <Text inline>
-                {message}
-                {helperLink ? (
-                  <Fragment>
-                    {' '}
-                    <HelperLink {...helperLink} />
-                  </Fragment>
-                ) : null}
-              </Text>
+  const textContent = (
+    <Flex.Item flex="grow">
+      <Box dangerouslySetInlineStyle={{ __style: !isDefault ? { marginTop: '-1px' } : {} }}>
+        {typeof message === 'string' ? (
+          <Text inline>
+            {message}
+            {helperLink ? (
+              <Fragment>
+                {' '}
+                <HelperLink {...helperLink} />
+              </Fragment>
             ) : null}
-            {typeof message !== 'string' && Children.only(message).type.displayName === 'Text'
-              ? message
-              : null}
-          </Box>
-        </Flex.Item>
-
-        {shouldShowButtons && (
-          <Flex.Item flex="none">
-            <Flex alignItems="center" gap={{ row: 4, column: 0 }}>
-              {primaryAction && (
-                <Box display="none" mdDisplay="flex" flex="none">
-                  <PrimaryAction {...primaryAction} />
-                </Box>
-              )}
+          </Text>
+        ) : null}
+        {typeof message !== 'string' && Children.only(message).type.displayName === 'Text'
+          ? message
+          : null}
+      </Box>
+    </Flex.Item>
+  );
 
-              {dismissButton && <DismissButton {...dismissButton} />}
-            </Flex>
-          </Flex.Item>
+  const mdActions = shouldShowButtons ? (
+    <Flex.Item flex="none">
+      <Flex alignItems="center" gap={{ row: 4, column: 0 }}>
+        {primaryAction && (
+          <Box display="none" mdDisplay="flex" flex="none">
+            <PrimaryAction {...primaryAction} />
+          </Box>
         )}
+
+        {dismissButton && <DismissButton {...dismissButton} />}
       </Flex>
-      {!isBare && primaryAction && (
-        <Box display="flex" mdDisplay="none" flex="none" alignSelf="end" marginTop={4}>
-          <PrimaryAction {...primaryAction} />
-        </Box>
-      )}
+    </Flex.Item>
+  ) : null;
+
+  const actions = !isBare && primaryAction && (
+    <Box display="flex" mdDisplay="none" flex="none" alignSelf="end" marginTop={4}>
+      <PrimaryAction {...primaryAction} />
     </Box>
   );
+
+  return (
+    <Fragment>
```

Comment:
This method is going to create issues for testing by duplicating all the elements in the DOM. Can we do this instead with CSS? I'll reach out tomorrow, but let's not merge this yet

Reply:
Ok... le me exploreee